### DISCUSSION TITLE
Generation from partially done JSON/List

### DISCUSSION THREAD

**Discussion by isamu-isozaki** [outlines-dev/outlines#694] [Q&A] [answered]
What I want is to complete a list of BaseTasks given couple BaseTasks are already defined like `[BaseTask1, BaseTask2`
and I want the model to complete the list. The schema is below

```python
class CompletionStatus(str, Enum):
    todo = 'todo'
    done = 'done'
    inprogress = "inprogress"

class BaseTask(BaseModel):
    status: CompletionStatus
    task_description: constr(max_length=50)
```

**Comment by rlouf** [comment, marked as answer]
If I understand correctly you can already do this, and it could be handled more elegantly once #667 is implemented. For instance:

```python
gen_choice = generate.choice(model, [", ", "]"])
gen_task = generate.json(model, BaseTask)

prompt = "[BaseTask1, BaseTask"
while True:
    result = gen_choice(prompt)
    if result == "]":
        return prompt + "]"
    else:
        prompt += ", "
        result = gen_task(prompt)
        prompt += result
```

**Reply by isamu-isozaki** [reply]
Thanks! Yeah, now that I think about it this is exactly what I wanted. Though yes, a more general solution might be cool
